flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/PreValidateReWriter.scala

```diff
@@ -234,13 +234,23 @@ object PreValidateReWriter {
 
     call.getKind match {
       case SqlKind.SELECT =>
+        if (
+          targetPosition
+            .size() != 0 && call.asInstanceOf[SqlSelect].getSelectList.size() != targetPosition
```

Comment:
to improve code readability:
- you can store `call.asInstanceOf[SqlSelect]` in a local var because its needed again below
- also `targetPosition.isEmpty` is shorter

This should make the if more readable and fit into one line